tools/analyzer_plugin/lib/src/diagnostic/boilerplate_validator.dart

```diff
@@ -67,6 +87,8 @@ class BoilerplateValidatorDiagnostic extends DiagnosticContributor {
       // Do not lint anything that is not a likely boilerplate member that will actually get generated.
       if (member.versionConfidences.toList().every((vcp) => vcp.confidence <= Confidence.neutral)) continue;
 
+      if (isPart(result.unit)) continue;
```

Comment:
Because of this, there's no error if components are declared in parts and there's no generated import in the main library.

Sorry, I can't remember; we going to handle that case in a separate PR?

Reply:
For posterity: we'll handle this separately